services/resource-usage-tracker/tests/unit/with_dbs/test_collect_container_resource_usage_task__on_update_set.py

```diff
@@ -0,0 +1,147 @@
+
+_END_DATETIME = 1687132800
+_START_DATETIME = 1687046400
+_NUM_OF_GENERATED_OUTPUTS = 20
+
+
+def _update_variable_if_smaller(existing_value, new_value):
+    if existing_value is None or new_value < existing_value:
+        existing_value = new_value
+    return existing_value
+
+
+def _update_variable_if_bigger(existing_value, new_value):
+    if existing_value is None or new_value > existing_value:
+        existing_value = new_value
+    return existing_value
+
+
+@pytest.fixture
+def random_promql_output_generator():
+    random_seed = random.randint(0, 100)
+    _logger.info("Random seed %s", random_seed)
+    Faker.seed(random_seed)
+    faker = Faker()
+
+    generated_data: list = []
+    min_timestamp_value: int | None = None
+    max_timestamp_value: int | None = None
+    max_float_value: float | None = None
+
+    for _ in range(_NUM_OF_GENERATED_OUTPUTS):
+        a = faker.unix_time(end_datetime=_END_DATETIME, start_datetime=_START_DATETIME)
+        b = faker.unix_time(end_datetime=_END_DATETIME, start_datetime=_START_DATETIME)
+        (smaller_timestamp, bigger_timestamp) = (a, b) if a < b else (b, a)
+
+        random_float = faker.pyfloat(
+            positive=True, min_value=0.157543565, max_value=500
+        )
+
+        min_timestamp_value = _update_variable_if_smaller(
+            min_timestamp_value, smaller_timestamp
+        )
+        max_timestamp_value = _update_variable_if_bigger(
+            max_timestamp_value, bigger_timestamp
+        )
+        max_float_value = _update_variable_if_bigger(max_float_value, random_float)
+
+        data_point = {
+            "metric": {
+                "container_label_com_docker_compose_oneoff": "False",
+                "container_label_com_docker_compose_project_working_dir": "/tmp/tmp_3seh6kp",
+                "container_label_com_docker_compose_version": "1.29.1",
+                "container_label_product_name": "osparc",
+                "container_label_simcore_service_settings": '[{"name": "ports", "type": "int", "value": 8888}, {"name": "env", "type": "string", "value": ["DISPLAY=:0"]}, {"name": "env", "type": "string", "value": ["SYM_SERVER_HOSTNAME=sym-server_%service_uuid%"]}, {"name": "mount", "type": "object", "value": [{"ReadOnly": true, "Source": "/tmp/.X11-unix", "Target": "/tmp/.X11-unix", "Type": "bind"}]}, {"name": "constraints", "type": "string", "value": ["node.platform.os == linux"]}, {"name": "Resources", "type": "Resources", "value": {"Limits": {"NanoCPUs": 4000000000, "MemoryBytes": 17179869184}, "Reservations": {"NanoCPUs": 100000000, "MemoryBytes": 536870912, "GenericResources": [{"DiscreteResourceSpec": {"Kind": "VRAM", "Value": 1}}]}}}]',
+                "container_label_simcore_user_agent": "puppeteer",
+                "container_label_study_id": "52d7e1a8-0c27-11ee-bec2-02420a0b0fc7",
+                "container_label_user_id": "43820",
+                "container_label_uuid": "2b231c38-0ebc-5cc0-9030-1ffe573f54e9",
```

Comment:
please randomize these uuids if not already so

Reply:
As this service/or tasks that will run in them, can be run anytime (for example when there is 3 hours outage ->  we can run this task in the time range when the outage took place. Therefore this test specifically tests whether values: created timestamp, last Prometheus scraped timestamp, and generated float value for total CPU usage seconds in *the same container* are correctly imported/updated in the database even though we run the task in different time slots.

Reply:
As this service/or tasks that will run in them, can be run anytime (for example when there is 3 hours outage -> we can run this task in the time range when the outage took place. Therefore this test specifically tests whether values: created timestamp, last Prometheus scraped timestamp, and generated float value for total CPU usage seconds in the same container are correctly imported/updated in the database even though we run the task in different time slots.

Reply:
I think I did not express my concern precise enough:, dont ever push or merge UUIDs of actual projects pr services into the open-source simcore repo. If not already done, manually change at least 6 chars please

Reply:
;)

Reply:
ok got it :+1: uuid randomized